### DISCUSSION TITLE
Should we keep supporting 1.13 - 1.15?

### DISCUSSION THREAD

**Discussion by TPGamesNL** [SkriptLang/Skript#4625] [Suggestions] [answered]
Recently, the question has come up if we should keep supporting the server versions 1.13 through 1.15. 

These versions aren't used much (between 1.6% and 3.6%):
![image](https://user-images.githubusercontent.com/29547183/155532215-b3f013fe-f94f-4b26-a9d1-1b8d69ab8a46.png)

Dropping support for them would allow us to switch to a higher Java version for Skript's runtime without problems, since Minecraft versions 1.16 and above already require at least Java 16, https://github.com/SkriptLang/Skript/issues/1321.

**Comment by Moderocky** [comment]
We could create a version support policy to solve both this and future update issues.

My recommendation would be that each new Skript release (excluding alphas and patches) could support only the last three major Minecraft versions, and the fourth could get 'end-of-life' support.

As an example for my suggestion: 
> This would mean that for 2.6.2 we would provide new features, improvements and fixes for the past 3 majors (1.18, 1.17, 1.16) and only fixes for the one before that (1.15) since 2.6.2 would be the last 1.15-supporting version.
We could, for 2.6.2, choose to drop support below Java 11 (the previous LTS) but this wouldn't be necessary.
Then, for 2.6.3, Minecraft 1.15-support would be phased out entirely and we would be free to update the required Java version to 15 or 16 (I think this is the maximum supported by Minecraft 1.16.)
When 1.19 is released, we would do an end-of-life version for Minecraft 1.16 before dropping this (and bumping the Java version to 17 LTS.)

By setting a policy for minecraft versions we're also solving the Java version question. Three major minecraft versions seems to be enough to capture a significant portion of the user-base (over two thirds) and with the fourth getting end-of-life support that gives people extra warning to update and reduces security concerns for people who linger on old versions for whatever reason.

If/when we set up a proper org homepage, this could display the latest supported Skript version for each Minecraft version to help users find the right one, as well as providing information about reporting bugs or making suggestions.

**Comment by ShaneBeee** [comment]
In my opinion, like I do with SkBee, the last 3 major versions should be the ONLY supported versions.
IE: right now 1.16.x, 1.17.x and 1.18.x

Anyone using anything older just makes no sense.
Take 1.15.2, why in the world would you be using this?
1.12.2 I sort of understand, same with 1.8.8... neither of those we support but they are both used more than 1.15, so 1.15.2 and older should be dropped, and Id say stick with the latest 3 major releases. 

Even 1.16 at this point seems odd to be running, but the last 2 versions didnt really have any major breaking API changes, so supporting 3 shouldn't be too difficult. If another version comes out, like say 1.19, with major breaking API changes, we can rethink at that point what we need to do with older version support.

tldr; Let's just support the last 3 major versions, nothing else is really needed.

**Reply by Lotzyprod** [reply]
Good idea

**Reply by Psekk** [reply]
Sounds like a solid plan.

**Reply by brodycritchlow** [reply]
I agree with this

**Reply by Psekk** [reply]
To add to this: this also allows for the quicker adaptation of new features

**Comment by Moderocky** [comment, marked as answer]
From Skript 2.10, we intend to set a policy of supporting only the last three major Minecraft versions.
For example, if the current Minecraft version was 1.21, we would support only 1.21, 1.20 and 1.19.

This doesn't mean we'll actively exclude older versions (we're not going to write some special code that prevents Skript starting up) but we're also not going to make any accommodations or work-arounds for them. If they don't work then they don't work.
